### PR TITLE
Add setFlickrAPIKey and getFlickrAPIKey + refactor/reformat code

### PR DESCRIPTION
**Pull request by elipousson**
Hello @koki25ando – this is an extensive pull request but hopefully a welcome one. This includes only limited changes to the package functionality but refactors the code for more consistent formatting and drops stringr as an imported package.

The most important feature additions are the setFlickrAPIKey and getFlickrAPIKey functions. The latter function is not exported but they are based on similar functions from the [mapboxapi package](https://github.com/walkerke/mapboxapi) and make it easy to store a Flickr API key as an environment variable and then load that key as needed to call the functions.

Let me know if you have any suggestions! I'd be happy to incorporate an update to the new version of pkgdown and add a few tests as part of this same pull request if that is helpful.

### CODE REVIEW

**Comment by koki25ando** [comment]
@elipousson  WOW. 👀
first of all... BIG THANKS for your contributions...! 
I will definitely check out and look through your codes. They are already looking super cool to me, especially 'setFlickrAPIKey', 'getFlickrAPIKey '.

right now please give me some time to look through and review them. I will give you updates as soon as possible.

Your contributions are always welcome!!! 🥳
thanks, Koki

File: R/getExif.R [hunk 1]
```diff
@@ -1,27 +1,38 @@
 #' Returns a data of EXIF data for a given photo.
 #'
-#' Return a data of Exif data for a given photo. The calling user must have permission to view the photo.
+#' Return a data of EXIF data for a given photo. The calling user must have
+#' permission to view the photo.
```
Comment (koki25ando) [inline R/getExif.R:4]: thanks for all these reformattings... much appreciated!

**Comment by elipousson** [comment]
Following up to confirm that I added in the update to the pkgdown site (the new Bootstrap 5 theme is very nice) and updated the vignette so it works with the new API utility functions.

**Comment by elipousson** [comment]
One last minor feature change: renaming licence_id to license_id to match the spelling used by the API. Would you mind if I also add myself as a contributor under the package description?

**Comment by koki25ando** [comment]
> Would you mind if I also add myself as a contributor under the package description?

Not at all! Please add your name!👍

File: R/getPhotoInfo.R [hunk 2]
```diff
@@ -1,59 +1,67 @@
...
+
+  output <- match.arg(tolower(output), c("all", "location", "date", "url", "tags"))
+
+  output_data <- switch(output,
+    "all" = data$photo,
+    "location" = janitor::clean_names(as.data.frame(data$photo$location)),
+    "data" = janitor::clean_names(as.data.frame(data$photo$dates)),
+    "url" = janitor::clean_names(as.data.frame(data$photo$urls$url)),
+    "tags" = janitor::clean_names(as.data.frame(data$photo$tags$tag))
+  )
```
Comment (koki25ando) [inline R/getPhotoInfo.R:44]: much simpler... cool update

**Comment by elipousson** [comment]
That last commit has a one last significant change: switching over to use the httr2 package to make the request. I created a new generic function `FlickrAPIRequest` that all of the other functions now call. I think this actually should make it much easier to incorporate more functionality from the API into the package in the future. I also updated the README and added myself to the contributors. Let me know if you spot any issues!

File: DESCRIPTION [hunk 3]
```diff
@@ -1,20 +1,23 @@
 Package: FlickrAPI
 Title: Access to Flickr API
 Version: 0.1.0.1
-Authors@R: person("Koki", "Ando", email = "koki.25.ando@gmail.com", role = c("aut", "cre"))
+Authors@R: c(
+    person("Koki", "Ando", email = "koki.25.ando@gmail.com", role = c("aut", "cre")),
+    person("Eli", "Pousson", role = c("aut", "ctb"))
+    )
```
Comment (koki25ando) [inline DESCRIPTION:7]: 👍

File: README.Rmd [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+You can also search photos by tag and license.
+
+```{r}
+photo_search <- getPhotoSearch(
+  sort = "date-taken-desc",
+  tags = c("cats", "dogs"),
+  per_page = 50)
+
+knitr::kable(photo_search[1:4,])
+```
```
Comment (koki25ando) [inline README.Rmd:69]: thank you for adding these tutorial example codes!
these are really cool👍

**Comment by koki25ando** [comment]
LGTM!! Thanks for your contribution!!👍

**Comment by elipousson** [comment]
Are you interested or already planning to update the version on CRAN? Happy to assist with that if you'd like. I want to Import FlickrAPI into another package that I'm working on and I think it tries to use the CRAN automatically. There is probably a work around but seems like it would be nice to get the updated version out there.

**Pull request merged**